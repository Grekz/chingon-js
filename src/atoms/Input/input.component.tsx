import styled from '@emotion/styled'
import * as React from 'react'

interface IInputProps {
  type: InputTypes
  value?: string
  placeholder?: string
  label: string
}

type InputTypes = 'text' | 'number' | 'password' | 'email'

const Input = ({ value, placeholder, label }: IInputProps) => (
  <InputWrapper>
    {label ? <label>{label}</label> : null}
    <input type="text" value={value} placeholder={placeholder} />
  </InputWrapper>
)

const InputWrapper = styled.div`
  position: relative;
  display: flex;
  flex-wrap: wrap;
  align-items: stretch;
  width: 100%;
  > label {
    display: flex;
    align-items: center;
    padding: 0.375rem 0.75rem;
    margin-bottom: 0;
    font-size: 1rem;
    font-weight: 400;
    line-height: 1.5;
    color: #495057;
    text-align: center;
    white-space: nowrap;
    background-color: #e9ecef;
    border: 1px solid #ced4da;
    border-radius: 0.25rem;
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
  }
  > input {
    position: relative;
    flex: 1 1 auto;
    width: 1%;
    margin-bottom: 0;
    display: block;
    height: calc(1.5em + 0.75rem + 2px);
    padding: 0.375rem 0.75rem;
    font-size: 1rem;
    font-weight: 400;
    line-height: 1.5;
    color: #495057;
    background-color: #fff;
    background-clip: padding-box;
    border: 1px solid #ced4da;
    border-radius: 0.25rem;
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
    transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  }
`

export default Input
