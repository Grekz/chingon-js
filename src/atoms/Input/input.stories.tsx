import * as React from 'react'

// tslint:disable-next-line: no-implicit-dependencies
import { storiesOf } from '@storybook/react'

import Input from '.'

storiesOf('Atoms|Input', module).add('As Default', () => (
  <Input type="text" placeholder="Input text" label="Test label" />
))
