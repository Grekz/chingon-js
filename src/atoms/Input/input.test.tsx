import { configure, shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import serializer, { matchers } from 'jest-emotion'
import * as React from 'react'
import Input from '.'

configure({ adapter: new Adapter() })
expect.extend(matchers)
expect.addSnapshotSerializer(serializer)

describe('<Input /> rendering', () => {
  it('should render Input', () => {
    const wrapper = shallow(
      <Input type="text" placeholder="Input text" label="Test label" />
    )
    expect(wrapper).toHaveLength(1)
  })
  it('should match snapshot', () => {
    const tree = shallow(
      <Input type="text" placeholder="Input text" label="Test label" />
    )
    expect(toJson(tree)).toMatchSnapshot()
  })
})
