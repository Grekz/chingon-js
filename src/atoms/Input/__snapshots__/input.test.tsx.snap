// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> rendering should match snapshot 1`] = `
<Styled(div)>
  <label>
    Test label
  </label>
  <input
    placeholder="Input text"
    type="text"
  />
</Styled(div)>
`;
